libs/utils/env.py

```diff
@@ -525,6 +525,7 @@ def _init_target(self, force = False):
             self.target = devlib.AndroidTarget(
                     platform = platform,
                     connection_settings = self.__connection_settings,
+                    working_directory = '/data/local/tmp/devlib-target',
```

Comment:
While looking at this, I remembered we had already a 'workdir' configuration option, which is configurable via target.conf for example, which AFAIR was there exactly for that scope... however, perhaps due to some refactoring, it's now defined but never used :-1: 

You can see it's definition at line 224 above.

Can you please wire this up to this existing configuration option?

Mind that there is also a WORKING_DIR_DEFAULT, which I don't like so much.... we should probably instead have something like:
```python
elif platform_type.lower() == 'android':
        # Workaround for ARM-software/devlib#225
        self.workdir = conf.get('workdir', '/data/local/tmp/devlib-target')
        self._log.debug('Setup ANDROID target...')
        self.target = devlib.AndroidTarget(...)
#...
else:
             raise ValueError('Config error: not supported [platform] type {}'\
                     .format(platform_type))

self.workdir = self.target.working_directory
```
where this last statement is just to ensure to always have the two variables aligned.

Reply:
In the example above... of course in the Android case you pass in `self.workdir` as `working_directory` in the AndroidTarget ctor ;-)

Reply:
Devlib already defines default working directories for each of its target types - could I just drop the `WORKING_DIR_DEFAULT` thing ? That way I can just make `self.workdir` default to `None` and let devlib handle it.